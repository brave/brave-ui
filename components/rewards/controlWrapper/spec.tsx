/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import ControlWrapper from './index'

describe('ControlWrapper tests', () => {
  const baseComponent = (props?: object) => <ControlWrapper id='wrapper' {...props} />

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent()
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent())
      const assertion = wrapper.find('#wrapper').length
      expect(assertion).toBe(1)
    })
  })
})
