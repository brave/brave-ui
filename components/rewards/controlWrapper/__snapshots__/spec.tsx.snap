// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlWrapper tests basic tests matches the snapshot 1`] = `
.c0 {
  max-width: 254px;
  width: 100%;
  margin-bottom: 12px;
}

<div
  className="c0"
  id="wrapper"
/>
`;
